ostd/src/mm/kspace/kva.rs

```diff
@@ -0,0 +1,287 @@
+    pub fn writer(&'a self) -> VmWriter<'a> {
+        unsafe { VmWriter::from_kernel_space(self.var.start as *mut u8, self.var.len()) }
+    }
+}
+
+impl VmIo for KvaInner {
+    // `VmIo` counterparts
+    fn read_bytes(&self, offset: usize, buf: &mut [u8]) -> Result<()> {
+        // Do bound check with potential integer overflow in mind
+        let max_offset = offset.checked_add(buf.len()).ok_or(Error::Overflow)?;
+        if max_offset > self.var.len() {
+            return Err(Error::InvalidArgs);
+        }
+        let len = self.reader().skip(offset).read(&mut buf.into());
+        debug_assert!(len == buf.len());
+        Ok(())
+    }
+
+    fn write_bytes(&self, offset: usize, buf: &[u8]) -> Result<()> {
+        // Do bound check with potential integer overflow in mind
+        let max_offset = offset.checked_add(buf.len()).ok_or(Error::Overflow)?;
+        if max_offset > self.var.len() {
+            return Err(Error::InvalidArgs);
+        }
+        let len = self.writer().skip(offset).write(&mut buf.into());
+        debug_assert!(len == buf.len());
+        Ok(())
+    }
+}
+
+// static KVA_FREELIST: SpinLock<BtreeMap<Vaddr, KvaFreeNode>> = SpinLock::new(BtreeMap::new(KvaFreeNode::new(kspace::TRACKED_MAPPED_PAGES_RANGE)));
+pub static KVA_FREELIST: SpinLock<BTreeMap<Vaddr, KvaFreeNode>> = SpinLock::new(BTreeMap::new());
+
+pub struct Kva(KvaInner);
+
+impl Kva {
+    // static KVA_FREELIST: SpinLock<BtreeMap<Vaddr, KvaFreeNode>> = SpinLock::new(BtreeMap::new(KvaFreeNode::new(kspace::KVMALLOC_START_VADDR, kspace::KVMALLOC_RANGE)));
+
+    pub fn alloc(size: usize) -> Self {
+        let mut lock_guard = KVA_FREELIST.lock();
+        let inner = KvaInner::alloc(lock_guard.deref_mut(), size).unwrap();
+        Kva(inner)
+    }
+
+    pub fn start(&self) -> Vaddr {
+        self.0.var.start
+    }
+
+    pub fn end(&self) -> Vaddr {
+        self.0.var.end
+    }
+
+    /// Map pages into the kernel virtual area.
+    /// # Safety
+    /// The caller should ensure either the mapped pages or the range to be used doesn't
+    /// violate the memory safety of kernel objects.
+    pub unsafe fn map_pages<T: PageMeta>(&mut self, range: Range<Vaddr>, pages: Vec<Page<T>>) {
+        assert!(range.len() == pages.len() * PAGE_SIZE);
+        let page_table = KERNEL_PAGE_TABLE.get().unwrap();
+        let prop = PageProperty {
+            flags: PageFlags::RW,
+            cache: CachePolicy::Writeback,
+            priv_flags: PrivilegedPageFlags::GLOBAL,
+        };
+        // page_table
+        //         .map(&range, &(pages.start_paddr()..pages.start_paddr()+pages.len()), prop);
+        let mut cursor = page_table.cursor_mut(&range).unwrap();
+        for page in pages.into_iter() {
+            cursor.map(page.into(), prop);
+        }
+        tlb_flush_addr_range(&range);
+    }
+    /// Get the type of the mapped page.
+    pub unsafe fn unmap_pages(&mut self, range: Range<Vaddr>) {
```

Comment:
The comments does not match with the function. Please implement `unmap` in `KvaInner` and call the inner implementation in this API.